addons/hr_holidays/models/hr_employee_base.py

```diff
@@ -203,7 +203,11 @@ def _get_consumed_leaves(self, leave_types, target_date=False, ignore_future=Fal
                     leave_duration = leave[leave_duration_field]
                     skip_excess = False
 
-                    if sorted_leave_allocations.filtered(lambda alloc: alloc.allocation_type == 'accrual') and leave.date_from.date() > target_date:
+                    if leave.date_from.date() > target_date and sorted_leave_allocations.filtered(lambda a:
+                        a.allocation_type == 'accrual' and
+                        (a.date_to >= target_date or not a.date_to) and
```

Comment:
```suggestion
                        (not a.date_to or a.date_to >= target_date) and
```
Order is important here otherwise compare a bool and a date (see ci/runbot)